content/en/blog/_posts/2025-04-23-Kubernetes-v1-33-Release/index.md

```diff
@@ -0,0 +1,471 @@
+the default Merge policy maintains backward compatibility by combining specified groups with those from the container image's `/etc/group` file, whereas the new Strict policy applies only to explicitly defined groups.
+
+This enhancement helps to address security concerns where implicit group memberships from container images could lead to unintended file access permissions and bypass policy controls.
+
+This work was done as part of [KEP-3619: Fine-grained SupplementalGroups control](https://kep.k8s.io/3619) led by SIG Node.
+
+### Support for mounting images as volumes
+
+Support for using Open Container Initiative (OCI) images as volumes in Pods, introduced in v1.31, has graduated to beta. This feature allows users to specify an image reference as a volume in a Pod while reusing it as a volume mount within containers. It opens up the possibility of packaging the volume data separately, and sharing them among containers in a Pod without including them in the main image, thereby reducing vulnerabilities and simplifying image creation.
+
+This work was done as part of [KEP-4639: VolumeSource: OCI Artifact and/or Image](https://kep.k8s.io/4639) led by SIG Node and SIG Storage.
+
+### Support for user namespaces within Linux Pods
+
+One of the oldest open KEPs as of writing is [KEP-127](https://kep.k8s.io/127), Pod security improvement by using Linux [User namespaces](/docs/concepts/workloads/pods/user-namespaces/) for Pods. This KEP was first opened in late 2016, and after multiple iterations, had its alpha release in v1.25, initial beta in v1.30 (where it was disabled by default), and is now part of v1.33, where the feature is available by default.
+
+This support will not impact existing Pods unless you manually specify `pod.spec.hostUsers` to opt in. As highlighted in the [v1.30 sneak peek blog](/blog/2024/03/12/kubernetes-1-30-upcoming-changes/), this is an important milestone for mitigating vulnerabilities.
+
+This work was done as part of [KEP-127: Support User Namespaces in pods](https://kep.k8s.io/127) led by SIG Node.
+
+### Pod `procMount` option
+
+The `procMount` option, introduced as alpha in v1.12, and off-by-default beta in v1.31, has moved to an on-by-default beta in v1.33. This enhancement improves Pod isolation by allowing users to fine-tune access to the `/proc` filesystem. Specifically, it adds a field to the Pod `securityContext` that lets you override the default behavior of masking and marking certain `/proc` paths as read-only. This is particularly useful for scenarios where users want to run unprivileged containers inside the Kubernetes Pod using user namespaces. Normally, the container runtime (via the CRI implementation) starts the outer container with strict `/proc` mount settings. However, to successfully run nested containers with an unprivileged Pod, users need a mechanism to relax those defaults, and this feature provides exactly that.
```

Comment:
This should probably note that using `procMount: Unmasked` *requires* pods to run with user namespaces by setting `hostUsers: false`

Reply:
This does mention about the use of user namespace, but only in the context of running "unprivileged containers inside the Kubernetes Pod using user namespaces". Do you think we should have a clear example of those two flags being used in combination? Or is `procMount` helpful in other contexts, and the use of user namespace may be optional?

As I'm not versed in this area, I'm finding it difficult to fully understand this myself. Perhaps that's a sign that this could do a bit of wording update, do you have some suggestion?

Reply:
I am going ahead with resolving this one -- I think this blog piece would help introducing the change, and more details can be found elsewhere including doc / KEP.